src/core.jl

```diff
@@ -501,6 +509,46 @@ function Base.close(sess::Session)
     return nothing
 end
 
+mutable struct DeviceList
+    ptr::Ptr{Cvoid}
+    count::Int
+
+    function DeviceList(s::Session)
+        status = Status()
+        ptr = @tfcall(:TF_SessionListDevices, Ptr{Cvoid},
+            (Ptr{Cvoid}, Ptr{Cvoid}), s, status)
+        check_status(status)
+        count = @tfcall(:TF_DeviceListCount, Cint, (Ptr{Cvoid},),
+            ptr)
+        this = new(ptr, count)
+        finalizer(this) do self
+            close(self)
+        end
+        this
+    end
+end
+
+struct DeviceInfo
+    name::String
+    device_type::String
+    membytes::Int64
+end
+
+Base.unsafe_convert(::Type{Ptr{Cvoid}}, l::DeviceList) = l.ptr
```

Comment:
Can we not use single letter variable names?
In particular not using `l` which often looks like `1` or `I`.